### ISSUE TITLE
Rule L045 Query defines CTE but does not use it fails lint on UPDATE

### ISSUE THREAD

**Issue by natethor** [sqlfluff/sqlfluff#3396]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When linting a SQL file that uses a CTE to then "join" during an UPDATE statement, sqlfluff fails with L045

### Expected Behaviour

I would expect sqlfluff to be able to see that the CTE created is referenced inside the following UPDATE statement.

### Observed Behaviour

```
sqlfluff lint --dialect postgres update_with_cte.sql
== [update_with_cte.sql] FAIL
L:   1 | P:   6 | L045 | Query defines CTE "cte" but does not use it.
All Finished 📜 🎉!
```

### How to reproduce

update_with_cte.sql
```sql
WITH cte AS (
    SELECT
        id,
        name,
        description
    FROM table1
)

UPDATE table2
SET
    name = cte.name,
    description = cte.description
FROM cte
WHERE table2.id = cte.id;
```

### Dialect

postgres

### Version

sqlfluff, version 0.13.2
Python 3.10.4

### Configuration

No custom config needed to reproduce this error.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
